.github/workflows/ci.yml

```diff
@@ -85,6 +85,37 @@ jobs:
         chmod +x codecov
         ./codecov
 
+  #
+  # Android build job
+  #
+  android:
+    runs-on: macos-latest
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip ci]')"
+
+    # Definition of the build matrix
+    strategy:
+      matrix:
+        java: [ 11 ]
+        android-api: [26, 27, 28, 29]
```

Comment:
@TWiStErRob so I have to remove 30-31 API levels (for now), the emulator GA seems to incorrectly deduct the arch:

```
/bin/sh -c \sdkmanager --install 'system-images;android-31;default;x86' --channel=0 > /dev/null
  Warning: Failed to find package 'system-images;android-31;default;x86'
```

The result is that emulator is not installed and the test run fails. I think we could hack the arch parameters or just get to that later on when it is fixed. I think it looks good now. Thanks!

Reply:
It just doesn't exist: `sdkmanager --list | grep android- | grep default`:
```
  system-images;android-26;default;arm64-v8a                                               | 1            | ARM 64 v8a System Image
  system-images;android-26;default;x86                                                     | 1            | Intel x86 Atom System Image
  system-images;android-26;default;x86_64                                                  | 1            | Intel x86 Atom_64 System Image
  system-images;android-27;default;arm64-v8a                                               | 1            | ARM 64 v8a System Image
  system-images;android-27;default;x86                                                     | 1            | Intel x86 Atom System Image
  system-images;android-27;default;x86_64                                                  | 1            | Intel x86 Atom_64 System Image
  system-images;android-28;default;arm64-v8a                                               | 1            | ARM 64 v8a System Image
  system-images;android-28;default;x86                                                     | 4            | Intel x86 Atom System Image
  system-images;android-28;default;x86_64                                                  | 4            | Intel x86 Atom_64 System Image
  system-images;android-29;default;arm64-v8a                                               | 8            | ARM 64 v8a System Image
  system-images;android-29;default;x86                                                     | 8            | Intel x86 Atom System Image
  system-images;android-29;default;x86_64                                                  | 8            | Intel x86 Atom_64 System Image
  system-images;android-30;default;arm64-v8a                                               | 1            | ARM 64 v8a System Image
  system-images;android-30;default;x86_64                                                  | 10           | Intel x86 Atom_64 System Image
  system-images;android-31;default;arm64-v8a                                               | 4            | ARM 64 v8a System Image
  system-images;android-31;default;x86_64                                                  | 4            | Intel x86 Atom_64 System Image

```
I would be very surprised if the "latest" MacOS, that GitHub just released recently would be x86.

Can you try adding the numbers and forcing the arch?

https://github.com/ReactiveCircus/android-emulator-runner/blob/e1248ae048e31ec2e171adc06be585d06c1f936e/action.yml#L14-L16

```diff
      uses: reactivecircus/android-emulator-runner@v2
      with:
        api-level: ${{ matrix.android-api }}
+       arch: x86_64 # or arm64-v8a (don't know if these Macs are M1/M2 or not)
```

Reply:
`x86_64` does not work for sure, I tried that before, let me try `arm`

Reply:
What does "does not work for sure" means, because in their examples they also use that:
https://github.com/ReactiveCircus/android-emulator-runner#usage--examples

Reply:
`PANIC: Avd's CPU Architecture 'arm64' is not supported by the QEMU2 emulator on x86_64 host.` sounds good.

Reply:
It seems like going up to `31` is fine, updated the matrix, after 31 - failures :(

Reply:
I think 33 would work, told you 32 is "special": https://github.com/mockito/mockito/pull/2893#discussion_r1093002196

Notice that in the above table there's no 32 and 33, but there are emulators.

Let's merge it as is, this is a huge improvement over "nothing". We can figure out 32/33 later.

Reply:
Oh sorry, I missed that the issue was only specific to 32

Reply:
Let's see why https://issuetracker.google.com/issues/267458959